service/debugger/debugger.go

```diff
@@ -916,7 +917,44 @@ func (d *Debugger) Goroutines(start, count int) ([]*api.Goroutine, int, error) {
 	if err != nil {
 		return nil, 0, err
 	}
+	if start > 0 || nextg >= 0 {
```

Comment:
I'm confused, here we're checking if `start > 0` and then trying to return all running goroutines, however in the commit and documentation it's specified that we only want to prioritize and return only running goroutines when `start == 0`. This leads us to having a bunch of conditionals below, I think we can clean this up by having this check first if `start ==0`, no?

Reply:
There are two paths here: `start == 0 && nextg >= 0` where we add running goroutines and `start > 0` where we remove them (because we returned them before when we had `start == 0`). It is a bit confusing as it is, I rewrote it by factoring out the common code and moving the conditions outside, hopefully it's clearer.